### ISSUE TITLE
Make support for "after column"

### ISSUE THREAD

**Issue by Casperhr** [vapor/mysql-driver#41]
Missing feature
```sql
ALTER TABLE `listings` ADD `client_id` INT NULL DEFAULT NULL AFTER `object_no`;
```

**Comment by tanner0101** [comment]
Looks neither SQLite or PostgreSQL actually support this. Should probably be added using the DatabaseSchema storage and specific extensions on SchemaBuilder where database type is MySQL.

**Comment by tanner0101** [comment]
Had to get this one done before it turned a year old ;) 

<details>
<summary>CustomOrder.swift</summary>

```swift
struct CustomOrder: MySQLModel {
    var id: Int?
    var a: Int
    var b: Int
}
```

</details>

<details>
<summary>CreateCustomOrder.swift</summary>

```swift
struct CreateCustomOrder: MySQLMigration {
    static func prepare(on conn: MySQLConnection) -> Future<Void> {
        return MySQLDatabase.create(CustomOrder.self, on: conn) { builder in
            builder.field(for: \.id)
            builder.field(for: \.a)
        }
    }
    
    static func revert(on conn: MySQLConnection) -> Future<Void> {
        return MySQLDatabase.delete(CustomOrder.self, on: conn)
    }
}
```
</details>

```swift
struct AddColumnBToCustomOrder: MySQLMigration {
    static func prepare(on conn: MySQLConnection) -> Future<Void> {
        return MySQLDatabase.update(CustomOrder.self, on: conn) { builder in
            builder.field(for: \.b)
            builder.order(\.b, after: \.id)
        }
    }
    
    static func revert(on conn: MySQLConnection) -> Future<Void> {
        return MySQLDatabase.update(CustomOrder.self, on: conn) { builder in
            builder.deleteField(for: \.b)
        }
    }
}
```

![screen shot 2018-06-21 at 7 46 52 pm](https://user-images.githubusercontent.com/1342803/41750826-05f2d38e-758c-11e8-9ac4-159cfc06df30.png)

This will be available in 3.0.0 RC 4.1
